src/operations/write.ts

```diff
@@ -267,7 +276,8 @@ const writeRoot = (
     if (typename === 'Mutation') {
       // We run side-effect updates after the default, normalized updates
       // so that the data is already available in-store if necessary
-      const updater = ctx.store.updates[fieldName];
+      const updater =
+        ctx.store.updates.Mutation && ctx.store.updates.Mutation[fieldName];
```

Comment:
Shall we update the type and defaulting so we don't have to check for `Mutation` here?

Reply:
If we update the type we tell the user that it it's mandatory to include updates.Subscription and updates.Mutation even though one of those two could be not needed. Up to you, then we need an invariant at initialization to check

Reply:
I think we should allow them to be optional for the user (`Partial<UpdatesConfig>`) but otherwise required fields that we can default for. Sounds reasonable to me but I'm not sure.

We could also consider adding `updates[typename]` for anything, but I'm not sure if that makes sense. It may be useful to make more generic updates like (`Todo: () => {/* add to a big todos list query */}`)

Reply:
I like the seperation a bit more personally